File: Tree/TreeHeight.py
class Node:
	def __init__(self, value):
		self.value = value
		self.right = None
		self.left = None

def tree_height(root):
	height = 0

	def traverse(node, level):
		if node == None:
			return

		nonlocal height 
		height = max(height, level)

		traverse(node.left, level+1)
		traverse(node.right, level+1)

	traverse(root, height+1)

	return height

a = Node(1)
b = Node(2)
c = Node(3)
d = Node(4)
e = Node(5)
f = Node(6)
g = Node(7)

a.left = b
b.left = c
a.right = d
d.left = e
d.right = f
f.right = g

print(tree_height(a))

File: Tree/TreeHeight.js
function Node(value){
	this.value = value;
	this.left = null;
	this.right = null;
}

function treeHeight(root){

	var height = 0;

	function traverse(node, level){
		if(node === null){
			return;
		}

		height = Math.max(level, height);

		traverse(node.left, level+1);
		traverse(node.right, level+1);
	}

	traverse(root, height+1)

	return height;
}

var a = new Node(1);
var b = new Node(2);
var c = new Node(3);
var d = new Node(4);
var e = new Node(5);

a.left = b;
a.right = c;
b.left = d;
d.right = e;

console.log(treeHeight(a));